pyvista/core/filters/poly_data.py

```diff
@@ -2601,3 +2601,114 @@ def strip(poly_data, join=False, max_length=1000, pass_cell_data=False,
         alg.SetPassThroughPointIds(pass_point_ids)
         _update_alg(alg, progress_bar, 'Stripping Mesh')
         return _get_output(alg)
+
+    def collision(poly_data, other_mesh, contact_mode=0, box_tolerance=0.001,
+                  cell_tolerance=0.0, n_cells_per_node=2, generate_scalars=False,
+                  progress_bar=False):
+        """Perform collision determination between two polyhedral surfaces.
+
+        If ``collision_mode`` is set to all contacts, the output will
+        be lines of contact. If ``collision_mode`` is first contact or half
+        contacts then the Contacts output will be vertices.
+
+        Parameters
+        ----------
+        other_mesh : pyvista.PolyData
+            Other mesh to test collision with.
+
+        contact_mode : int, optional
+            Contact mode.  One of the following:
+
+            * 0 - All contacts. Find all the contacting cell pairs
+              with two points per collision
+            * 1 - First contact. Find all the contacting cell pairs
+              with one point per collision.
+            * 2 - Half contacts. Quickly find the first contact point.
```

Comment:
Is there such a thing as "first contact point" in this implementation?  Maybe I misunderstand exactly what multiple collisions means.

Reply:
> I think the descriptions for First and Half contacts are reversed here.

Quite right.

> Is there such a thing as "first contact point" in this implementation?

According to the header file and their documentation [vtkCollisionDetection](https://vtk.org/doc/nightly/html/classvtkCollisionDetectionFilter.html#details), yes.  As for the "how", I don't see any documentation on how it computes the collisions.